tests/rptest/services/redpanda.py

```diff
@@ -1821,12 +1827,6 @@ def metrics_sample(
             return None
         return MetricSamples(sample_values)
 
-    def read_configuration(self, node):
-        assert node in self._started
-        with self.config_file_lock:
-            with node.account.open(RedpandaService.NODE_CONFIG_FILE) as f:
-                return yaml.full_load(f.read())
-
```

Comment:
Nice! I think we also don't need `config_file_lock` anymore. Kind of surprising that we only use it here.